src/js/editor.js

```diff
@@ -1,23 +1,79 @@
+/* global classifyObj */
 
+// External dependencies.
+import { handleClick } from './helpers';
+
+const { __ } = wp.i18n;
+const { registerPlugin } = wp.plugins;
+const { PluginPostStatusInfo } = wp.editPost; // eslint-disable-line no-unused-vars
 const { subscribe, select, dispatch } = wp.data;
+const { Button } = wp.components; // eslint-disable-line no-unused-vars
+const { Fragment } = wp.element; // eslint-disable-line no-unused-vars
 
 let saveHappened = false;
 let showingNotice = false;
 
-subscribe( () => {
+/**
+ * Display error notice if classify failed.
+ */
+const showNotice = () => {
+	const meta = select( 'core/editor' ).getCurrentPostAttribute( 'meta' );
+	if ( meta._classifai_error ) {
+		showingNotice = true;
+		const error = JSON.parse( meta._classifai_error );
+		dispatch( 'core/notices' ).createErrorNotice( `Failed to classify content with the IBM Watson NLU API. Error: ${ error.code } - ${ error.message }` );
+		saveHappened = false;
+		showingNotice = false;
+	}
+};
 
+subscribe( () => {
 	if ( false === saveHappened ) {
 		saveHappened = true === wp.data.select( 'core/editor' ).isSavingPost();
 	}
 
 	if ( saveHappened && false === wp.data.select( 'core/editor' ).isSavingPost() && false === showingNotice ) {
-		const meta = select( 'core/editor' ).getCurrentPostAttribute( 'meta' );
-		if ( meta._classifai_error ) {
-			showingNotice = true;
-			const error = JSON.parse( meta._classifai_error );
-			dispatch( 'core/notices' ).createErrorNotice( `Failed to classify content with the IBM Watson NLU API. Error: ${ error.code } - ${ error.message }` );
-			saveHappened = false;
-			showingNotice = false;
-		}
+		showNotice();
 	}
 } );
+
+if ( 'true' === classifyObj.show_generate_button ) {
+	/**
+	 * Add option to generate tags for existing content.
+	 *
+	 * @returns {JSX.Element}
+	 * @constructor
+	 */
+	const ClassifAIAddGenerateTagsButton = () => (
+		<PluginPostStatusInfo>
+			<Fragment>
+				<Button
+					isSecondary={ true }
+					data-id= { select( 'core/editor' ).getCurrentPostId() }
+					onClick={ ( e ) => {
+						handleClick(
+							{
+								button: e.target,
+								endpoint: '/classifai/v1/generate-tags/',
+								callback: resp => {
+									if ( true === resp.success ) {
+										showNotice();
+										const isPostSaved = select( 'core/editor' ).didPostSaveRequestSucceed();
+										if ( true === isPostSaved ) {
+											window.location.reload();
```

Comment:
Can you please try using `editEntityRecord` to update tags without reloading the edit page?

`editEntityRecord` can be used to update taxonomy data.

Reply:
> Trying to update records based on generated terms, breaks the editor using following snippet.

@dinhtungdu I was getting an error when using `editEntityRecord`, but that was due to an invalid use, it is now working but only for posts that have some terms already attached to them, see below.

Existing post with some terms: 

https://user-images.githubusercontent.com/13589980/111116414-00c8b000-858c-11eb-8cc4-5fa6e2481fe7.mp4


Existing post with no terms doesn't work with above method, so a reload could be required in that case. We also don't have the taxonomy terms updated when generating tags on a new post in block editor.